### PR TITLE
feat: view scheduler job history component

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Component to view the history of the selected job. Also the items should be filtered according to the version.

## Related issues and discussion
#675 

## Screenshots, if any
Screenshot from 2020-08-24 00-16-08

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@abhaychawla  Please review the PR

File: src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        <td mat-cell *matCellDef="let jobHistoryData"> {{ jobHistoryData.status }} </td>
+      </ng-container>
+
+      <ng-container matColumnDef="run_type">
+        <th mat-header-cell *matHeaderCellDef mat-sort-header> Run Type </th>
+        <td mat-cell *matCellDef="let jobHistoryData">{{ jobHistoryData.triggerType }}</td>
+      </ng-container>
+
+      <ng-container matColumnDef="error_log">
+        <th mat-header-cell *matHeaderCellDef mat-sort-header> Error Log </th>
```
Comment (karantakalkar) [inline src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html:36]: refer to manage scheduler jobs component for this field.

File: src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   */
+  setJobHistory() {
+    this.dataSource = new MatTableDataSource(this.jobHistoryData.pageItems);
+    this.dataSource.paginator = this.paginator;
+    /** Search By Version */
+    this.dataSource.filterPredicate =
+    (data: any, filtersJson: string) => {
+      const matchFilter: any[] = [];
+      const filters = JSON.parse(filtersJson);
+
```
Comment (karantakalkar) [inline src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.ts:62]:
```suggestion
```

File: src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      filters.forEach((filter: any) => {
+        const val = data[filter.id] === null ? '' : data[filter.id];
+        if (filter.value !== '') {
+          matchFilter.push(val === parseInt(filter.value, 10));
+        } else if (filter.value === '') {
+          matchFilter.push(val.toString().toLowerCase().includes(filter.value.toLowerCase()));
+        }
+
```
Comment (karantakalkar) [inline src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.ts:70]:
```suggestion
```

File: src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  /**
+   * Filters data in scheduler Job History table based on passed value.
+   * @param {string} filterValue Value to filter data.
+   */
+  applyFilter(filterValue: string) {
+    const filterObject = [{
+      id: 'version',
+      value: filterValue
+    }];
+
```
Comment (karantakalkar) [inline src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.ts:43]:
```suggestion
```

File: src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+      <tr mat-row *matRowDef="let row; columns: displayedColumns;" class="select-row"></tr>
+
+    </table>
+
+    <mat-paginator [length]="jobHistoryData.totalFilteredRecords" [pageSize]="14"></mat-paginator>
+
+  </div>
+
+</div>
```
Comment (karantakalkar) [inline src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html:49]: new line at EOF

File: src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      </ng-container>
+
+      <ng-container matColumnDef="run_type">
+        <th mat-header-cell *matHeaderCellDef mat-sort-header> Run Type </th>
+        <td mat-cell *matCellDef="let jobHistoryData">{{ jobHistoryData.triggerType }}</td>
+      </ng-container>
+
+      <ng-container matColumnDef="error_log">
+        <th mat-header-cell *matHeaderCellDef mat-sort-header> Error Log </th>
+        <td mat-cell *matCellDef="let jobHistoryData"> </td>
```
Comment (muskankhedia) [inline src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html:37]: @BLasan Can you please make these changes?

The errorlog status needs to be displayed as in the community app which also displays a log modal on click.
Screenshot from 2020-05-17 11-47-54

Screenshot of  error log dailog box:
Screenshot from 2020-05-17 11-50-34

Comment (BLasan) [reply src/app/system/manage-scheduler-jobs/view-history-scheduler-job/view-history-scheduler-job.component.html:37]: I'll refactor them

**Comment by karantakalkar** [comment]
@BLasan can you update this?

**Comment by BLasan** [comment]
> @BLasan can you update this?

I'm really busy now. Can I take a look by tomorrow ?

**Comment by BLasan** [comment]
@karantakalkar I'm fixing them

**Comment by BLasan** [comment]
@karantakalkar Done

File: src/app/system/system.module.ts [hunk 7]
```diff
@@ -114,7 +115,18 @@ import { ConfigureMakerCheckerTasksComponent } from './configure-maker-checker-t
...
+  ],
+  entryComponents: [
+    ReportParameterDialogComponent,
+    AddEventDialogComponent,
+    ColumnDialogComponent,
+    EditHookComponent,
+    ViewHookComponent,
+    CreateHookComponent,
+    AddEventDialogComponent
+  ],
```
Comment (karantakalkar) [inline src/app/system/system.module.ts:130]:
```suggestion
 
```
Entry components declaration is no longer required.

Comment (BLasan) [reply src/app/system/system.module.ts:130]: I'll remove them all

**Comment by BLasan** [comment]
@karantakalkar Please review

**Review by karantakalkar — APPROVED** [review]

**Pull request merged**